src/utils/convert-to-cf-event.ts

```diff
@@ -11,13 +11,13 @@ export type IncomingMessageWithBodyAndCookies = IncomingMessage & {
 	cookies: Record<string, string>;
 };
 
-export function convertToCloudFrontEvent(req: IncomingMessageWithBodyAndCookies, config: CloudFrontConfig): CloudFrontRequestEvent {
+export function convertToCloudFrontEvent(req: IncomingMessageWithBodyAndCookies, config: CloudFrontConfig, customHeaders: Record<string, string>): CloudFrontRequestEvent {
 	const url = parse(req.url as string, false) as UrlWithStringQuery;
 	const request = {
 		clientIp: req.socket.remoteAddress as string,
 		method: req.method as string,
-		headers: toCloudFrontHeaders(req.headers),
-		uri: url.href as string,
+		headers: toCloudFrontHeaders({...customHeaders, ...req.headers}),
+		uri: url.pathname as string,
```

Comment:
Why the switch from `url.href` to `url.pathname`?

Reply:
The `url.href` contains the query string of the url. The `url.pathname` only contains the path without the query string which I think is the correct behaviour looking at https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/lambda-event-structure.html

Reply:
👌